components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRMService.java

```diff
@@ -925,4 +963,48 @@ private ServiceProvider cloneServiceProvider(ServiceProvider serviceProvider) {
         ServiceProvider clonedServiceProvider = gson.fromJson(gson.toJson(serviceProvider), ServiceProvider.class);
         return clonedServiceProvider;
     }
+
+    /**
+     * Validate SSA signature using jwks_uri.
+     * @param softwareStatement Software Statement
+     * @throws DCRMClientException
+     * @throws IdentityOAuth2Exception
+     */
+    private void validateSSASignature(String softwareStatement) throws DCRMClientException, IdentityOAuth2Exception {
+
+        SignedJWT signedJWT = null;
+        try {
+            signedJWT = SignedJWT.parse(softwareStatement);
+        } catch (ParseException e) {
+            throw new DCRMClientException(DCRMConstants.ErrorCodes.INVALID_SOFTWARE_STATEMENT,
+                    DCRMConstants.ErrorMessages.SIGNATURE_VALIDATION_FAILED.getMessage(), e);
+        }
+        String jwksURL = IdentityUtil.getProperty(SSA_VALIDATION_JWKS);
```

Comment:
At the beginning of the method, we have to check if the jwksURL is configured and if not skip the SSA signature validations (add a debug log as well).

Reply:
changed in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2178/commits/feb4731bf60075ddf78c6651b076e544ceea8add